src/runtime/audience-action-flow.ts

```diff
@@ -321,6 +327,50 @@ export class AudienceActionFlow {
   }
 
   /**
+   * Populates localStorage with PPS configuration parameters based on
+   * SurveyDataTransferRequest.
+   **/
+  private async storePpsValuesFromSurveyAnswers(
+    request: SurveyDataTransferRequest
+  ): Promise<void> {
+    const iabAudienceKey = StorageKeys.PPS_TAXONOMIES;
+    // PPS value field is optional and category may not be populated
+    // in accordance to IAB taxonomies.
+    const ppsConfigParams = request
+      .getSurveyQuestionsList()!
+      .flatMap((question) => question.getSurveyAnswersList())
+      .map((answer) => answer?.getPpsValue())
+      .filter((ppsValue) => ppsValue !== null);
+
+    const existingIabTaxonomy = await this.storage_.get(
+      iabAudienceKey,
+      /* useLocalStorage= */ true
+    );
+    try {
+      const existingIabTaxonomyValues =
+        JSON.parse(existingIabTaxonomy)?.[Constants.PPS_AUDIENCE_TAXONOMY_KEY]
+          ?.values || [];
+      const iabTaxonomyValues = Array.from(
+        new Set(ppsConfigParams.concat(existingIabTaxonomyValues))
+      );
+      const iabTaxonomy = {
+        [Constants.PPS_AUDIENCE_TAXONOMY_KEY]: {values: iabTaxonomyValues},
+      };
+
+      await Promise.resolve(
```

Comment:
Can we take it out of try/catch so that the current pps values are still stored even if reading/parsing existing signals fails?

Reply:
yeah, ended up making it fall back to a default value